stratum/hal/lib/barefoot/bfrt_packetio_manager.cc

```diff
@@ -115,9 +95,17 @@ ::util::Status BfrtPacketioManager::Shutdown() {
     packetout_header_.clear();
     packetin_header_size_ = 0;
     packetout_header_size_ = 0;
+    if (packet_receive_channel_) {
+      packet_receive_channel_->Close();
+    }
     initialized_ = false;
   }
-
+  {
+    absl::ReaderMutexLock l(&data_lock_);
+    if (rx_thread_initialized_) {
```

Comment:
I add a new variable called `rx_thread_initialized_`, the sde_rx_thread_id_` will be invalid if we push an invalid p4info/pipeline config. And `pthread_join` will cause the program to crash.
Not sure if we can solve this without adding this variable.

Reply:
Init `sde_rx_thread_id_` with 0 and check `if != 0` ?

Reply:
Also noticed that `sde_rx_thread_id_` is missing from the constructor init list.

Reply:
Another approach would be to spawn the thread at class creation, then `Shutdown` will never be called on failures. But that might not be logically possible.

Reply:
That would work, will do